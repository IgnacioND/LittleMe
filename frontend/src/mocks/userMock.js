export default {
  _id: '613e6e1a573916952b1e5f9e',
  name: '',
  username: '',
  email: '2@2.com',
  password: '1234',
  little: {
    name: 'Pepe',
    age: 1,
    evolution: 1,
    skin: 1,
    appeareance: {
      eyes: 1,
      mouth: 1,
      hair: 1,
      extra1: 1,
      extra2: 1,
      height: 1,
      width: 1,
    },
    stats: {
      money: 246,
      hunger: -24,
      hygiene: 1,
      mind: 6,
    },
    clothes: {
      shirt: 1,
      pants: 1,
      shoes: 1,
    },
    tendencies: {
      studies: 4,
      healthiness: 4,
      leisure: 2,
    },
  },
  house: [{
    name: 1,
    leftWall: 1,
    rightWall: 1,
    centerWall: 1,
    floor: 1,
    ceiling: 1,
    leftFurniture: 1,
    centerFurniture: 1,
    rightFurniture: 1,
    tidynessLevel: 1,
    _id: '613e6e1a573916952b1e5f9f',
  }],
  tasks: [
    '613e74af319c5d9cc291c71a',
    '613e750eff8cd48fd8c8d88f',
    '613e75437a2dabb296ffeada',
    '613e757e809a60a2a6c61fa9',
  ],
};
