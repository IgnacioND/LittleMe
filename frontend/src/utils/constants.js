export const MONEY_INCREMENT = 10;
export const STANDARD_INCREMENT = 1;
export const MONEYPLUS_INCREMENT = MONEY_INCREMENT * 2;
export const NULL_INCREMENT = 0;
