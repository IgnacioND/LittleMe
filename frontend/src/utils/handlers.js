// import { useDispatch } from 'react-redux';

export const goToTasksScreen = (navigation) => () => navigation.navigate('TasksScreen');
export const goToLoginScreen = (navigation) => () => navigation.navigate('LoginScreen');
export const goToSignupScreen = (navigation) => () => navigation.navigate('SignupScreen');
export const goToDashboardScreen = (navigation) => () => navigation.navigate('DashboardScreen');
export const goToProfileScreen = (navigation) => () => navigation.navigate('ProfileScreen');
export const goToCalendarScreen = (navigation) => () => navigation.navigate('CalendarScreen');
export const goToCharacterScreen = (navigation) => () => navigation.navigate('CharacterScreen');
export const goToCreateTaskScreen = (navigation) => () => navigation.navigate('CreateTaskScreen');
export const goToHomeScreen = (navigation) => () => navigation.navigate('HomeScreen');
export const goToScreen = (navigation) => () => navigation.navigate('Screen');
