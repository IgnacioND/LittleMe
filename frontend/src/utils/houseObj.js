const houseObj = [
  {
    name: 1,
    leftWall: 1,
    rightWall: 1,
    centerWall: 1,
    floor: 1,
    ceiling: 1,
    leftFurniture: 1,
    centerFurniture: 1,
    rightFurniture: 1,
    tidynessLevel: 1,
  },
];

module.exports = houseObj;
