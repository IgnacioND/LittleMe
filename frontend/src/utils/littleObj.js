const littleObj = {
  name: 'Pepe',
  age: 1,
  evolution: 1,
  skin: 1,
  appeareance: {
    eyes: 1,
    mouth: 1,
    hair: 1,
    extra1: 1,
    extra2: 1,
    height: 1,
    width: 1,
  },
  stats: {
    money: 3,
    hunger: 5,
    hygiene: 5,
    mind: 3,
  },
  clothes: {
    shirt: 1,
    pants: 1,
    shoes: 1,
  },
  tendencies: {
    studies: 1,
    healthiness: 1,
    leisure: 1,
  },
};
module.exports = littleObj;
