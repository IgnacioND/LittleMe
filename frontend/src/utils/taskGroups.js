const taskGroups = {
  work: ['reunion', 'meeting', 'colleage', 'worker', 'company',
    'coworker', 'co-worker', 'working', 'work', 'project'],
  studies: ['read', 'homework', 'book', 'learn', 'study', 'degree', 'master'],
  house: [
    'kitchen', 'living-room', 'room', 'bathroom',
    'buy', 'clean', 'cleaning', 'dormitory', 'hall',
    'cook', 'groceries', 'bleach', 'ironing', 'detergent',
    'laundry', 'dust', 'washing', 'floor', 'mop',
    'lawn', 'rubbish', 'tidy', 'vacuum', 'dishes', 'broom', 'chores'],
  leisure: ['friend', 'movie', 'movies', 'cinema', 'hangout',
    'hobby', 'holiday', 'relax', 'relaxation', 'rest', 'casual',
    'enjoy', 'travel', 'vacation', 'play', 'theater', 'hang out'],
  health: ['gym', 'dance', 'run', 'doctor', 'hair', 'football', 'play', 'dojo', 'fight',
    'lunch', 'medical', 'therapy', 'psycology', 'shrunk', 'eat', 'dinner', 'sleep',
    'breackfast', 'dinner', 'lunch'],

};
module.exports = taskGroups;
