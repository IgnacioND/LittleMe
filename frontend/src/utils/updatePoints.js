import {
  MONEY_INCREMENT,
  STANDARD_INCREMENT,
  MONEYPLUS_INCREMENT,
  NULL_INCREMENT,
} from './constants';

export default function updatePoints(task, completeUser) {
  const isWorkTask = task === 'work';
  const isHealthTask = task === 'health';
  const isLeisureTask = task === 'leisure';
  const isMindTask = ['study', 'house', 'leisure'].includes(task);
  const isHealthyTask = ['study', 'house', 'health'].includes(task);
  const isWorkStudyTask = ['study', 'work'].includes(task);
  const { stats, tendencies } = completeUser?.little || {};

  return {
    little: {
      stats: {
        money: stats?.money + (isWorkTask ? MONEYPLUS_INCREMENT : MONEY_INCREMENT),
        hunger: stats?.hunger + STANDARD_INCREMENT,
        hygiene: stats?.hygiene + (isHealthTask ? STANDARD_INCREMENT : NULL_INCREMENT),
        mind: stats?.mind + (isMindTask ? STANDARD_INCREMENT : NULL_INCREMENT),

      },
      tendencies: {
        studies: tendencies?.studies + (isWorkStudyTask ? STANDARD_INCREMENT : NULL_INCREMENT),
        healthiness: tendencies?.healthiness + isHealthyTask ? STANDARD_INCREMENT : NULL_INCREMENT,
        leisure: tendencies?.leisure + (isLeisureTask ? STANDARD_INCREMENT : NULL_INCREMENT),
      },
    },
  };
}
