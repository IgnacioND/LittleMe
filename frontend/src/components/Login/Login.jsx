import React from 'react';
import PropTypes from 'prop-types';
import {
  View, Text, Image, TextInput, TouchableOpacity,
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { validate } from 'validate.js';
import constraints from '../../utils/constraints';

import { loginUser } from '../../redux/actions/loggedUser.creator';

import generalStyles from '../styles/generalStyles';
import { goToHomeScreen } from '../../utils/handlers';

const appLogo = require('../../assets/images/logo.png');

export default function Login({ navigation }) {
  const dispatch = useDispatch();
  const [userEmail, setUserEmail] = React.useState('');
  const [userPassword, setUserPassword] = React.useState('');
  const [printWarning, setPrintWarning] = React.useState(false);

  const validationResult = validate({ emailAddress: userEmail }, constraints);

  const { email } = useSelector((store) => store.loggedUser);

  const loginHandler = () => {
    if (validationResult === undefined) {
      setPrintWarning(false);
      dispatch(loginUser({
        email: userEmail.toLowerCase(),
        password: userPassword,
      }));
    } else {
      setPrintWarning(true);
    }
  };

  const warningTexts = {
    none: '',
    wrongEmail: validationResult?.emailAddress[0],
    wrongAccount: 'Email or password are wrong',
  };

  let warningToPrint;
  if (printWarning) {
    warningToPrint = warningTexts.wrongEmail;
  } else if (email === 'erroneus') {
    warningToPrint = warningTexts.wrongAccount;
  } else { warningToPrint = warningTexts.none; }

  return (
    <View style={generalStyles.background}>
      <Image style={generalStyles.mainLogo} source={appLogo} />
      <View style={generalStyles.formContainer}>
        <TextInput placeholder="Username" style={generalStyles.normalInput} value={userEmail} onChangeText={(text) => setUserEmail(text)} />
        <Text testID="warningText">{warningToPrint}</Text>
        <TextInput placeholder="Login" secureTextEntry style={generalStyles.normalInput} value={userPassword} onChangeText={(text) => setUserPassword(text)} />
        <View style={generalStyles.textedButtonContainer}>
          <View style={generalStyles.textedButtonBack} />
          <TouchableOpacity style={generalStyles.textedButton} testID="loginButton" onPress={loginHandler}>
            <Text style={generalStyles.buttonText}>ENTER</Text>
          </TouchableOpacity>
        </View>
        <View style={generalStyles.textedButtonContainer}>
          <View style={generalStyles.textedButtonBack} />
          <TouchableOpacity style={generalStyles.textedButton} testID="backButton" onPress={goToHomeScreen(navigation)}>
            <Text style={generalStyles.buttonText}>BACK</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}

Login.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
