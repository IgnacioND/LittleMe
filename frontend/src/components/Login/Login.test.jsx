/* eslint-disable no-underscore-dangle */
import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import * as redux from 'react-redux';

import { validate } from 'validate.js';
import Login from './Login';
import { render } from '../../utils/test.utils';
import { loginUser } from '../../redux/actions/loggedUser.creator';
import '@testing-library/jest-dom/extend-expect';
import constraints from '../../utils/constraints';
import userMock from '../../mocks/userMock';

jest.mock('../../redux/actions/loggedUser.creator');
jest.mock('validate.js');

const navigation = {
  navigate: jest.fn(),
};

describe('Given a Login Component', () => {
  describe('When it is rendered', () => {
    let screen;

    beforeEach(() => {
      const initialState = {
        user: userMock,
      };
      screen = render(<Login navigation={navigation} />, initialState);
    });

    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });

    describe('And email recive an input', () => {
      test('Should set email', () => {
        const emailInput = screen.getByPlaceholderText('Username');

        fireEvent.changeText(emailInput, 'email12345678');
        expect(screen.queryByDisplayValue('email12345678')).not.toBe(null);
      });
    });
    describe('And password recive an input', () => {
      test('Should set password', () => {
        const passwordInput = screen.getByPlaceholderText('Login');
        const setPassword = jest.fn();
        fireEvent(passwordInput, 'onChangeText', setPassword);
        expect(setPassword).toHaveBeenCalled();
      });
    });
    describe('And user doesnt exist', () => {
      test('Should return not user warning', () => {
        jest
          .spyOn(redux, 'useSelector')
          .mockReturnValueOnce('erroneus');

        validate.mockReturnValue({ emailAddress: 'email12345678' }, constraints);
        loginUser.mockReturnValue({ type: '' });

        const loginButton = screen.getByTestId('loginButton');
        fireEvent.press(loginButton);
        const warningMessage = screen.getByTestId('warningText');
        expect(warningMessage).not.toBe(null);
      });
    });

    describe('When back button is pressed', () => {
      test('Should go to HomeScreen', () => {
        const backButton = screen.getByTestId('backButton');
        fireEvent.press(backButton);
        expect(navigation.navigate).toHaveBeenCalled();
      });
    });
  });
});
