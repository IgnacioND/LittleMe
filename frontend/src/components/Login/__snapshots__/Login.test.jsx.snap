// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Login Component When it is rendered Then screen should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFCA51",
      "flex": 1,
      "justifyContent": "space-around",
      "paddingBottom": 30,
      "paddingTop": 100,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "margin": 80,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 400,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Username"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderWidth": 2,
          "height": 45,
          "margin": 10,
          "padding": 10,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      testID="warningText"
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Login"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderWidth": 2,
          "height": 45,
          "margin": 10,
          "padding": 10,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "darkgrey",
            "borderColor": "black",
            "borderWidth": 2,
            "height": 30,
            "marginLeft": 3,
            "marginTop": 3,
            "width": 90,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFCA51",
            "borderColor": "black",
            "borderWidth": 2,
            "height": 30,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "width": 90,
          }
        }
        testID="loginButton"
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          ENTER
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "darkgrey",
            "borderColor": "black",
            "borderWidth": 2,
            "height": 30,
            "marginLeft": 3,
            "marginTop": 3,
            "width": 90,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFCA51",
            "borderColor": "black",
            "borderWidth": 2,
            "height": 30,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "width": 90,
          }
        }
        testID="backButton"
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          BACK
        </Text>
      </View>
    </View>
  </View>
</View>
`;
